zipline/algorithm.py

```diff
@@ -208,6 +208,9 @@ def __init__(self, *args, **kwargs):
         if not self.blotter:
             self.blotter = Blotter()
 
+        # Set the dt initally to the period start by forcing it to change
+        self.on_dt_changed(self.sim_params.period_start)
+
```

Comment:
This makes the self.datetime assignment on line 160 redundant, so you can delete it.